### PR TITLE
Wire : New "begin" method for late call configuration

### PR DESCRIPTION
**Pull request by jmchiappa**
**Summary**

This PR implements the following feature
`begin(uint8_t sda, uint8_t scl);`

**Why ?**
In some libraries, it requires a late call for SDA/SCL configuration. Have a look on u8g2 library. This method is supported by modern 32 bits architecture (ESP12, ESP8266).

**How to test ?**
1. Install u8g2 library
2. Open Graphic test
3. wire your OLED display to D6 (SCL) and D39 (SDA).
4. uncomment the line `U8G2_SSD1306_128X64_NONAME_F_HW_I2C u8g2(U8G2_R0, /* reset=*/ U8X8_PIN_NONE, /* clock=*/ 6, /* data=*/ 39);   // ESP32 Thing, HW I2C with pin remapping`

5. compile and test

**Known limitations**
This needs to add `defined(ARDUINO_ARCH_STM32)` in u8g2 library in order to take benefits of this feature. The PR will be submitted very soon.

### CODE REVIEW

File: libraries/Wire/src/Wire.h [hunk 1]
```diff
@@ -85,6 +85,7 @@ class TwoWire : public Stream {
       _i2c.sda = sda;
     };
     void begin(bool generalCall = false);
+    void begin(uint8_t, uint8_t);
```
Comment (fpistm) [inline libraries/Wire/src/Wire.h]:
```suggestion
    void begin(uint8_t, uint8_t);
```

**Comment by fpistm** [comment]
@jmchiappa 
Could you clean this PR to get only your change.
Thanks.

**Comment by jmchiappa** [comment]
> @jmchiappa
> Could you clean this PR to get only your change.
> Thanks.

@fpistm,
Sorry I don't know how to do that...

**Comment by fpistm** [comment]
@jmchiappa 
the main issue is that you made your PR on master. Then you pull the master.
So update you master to be inline with the original repo:
```
git remote add upstream https://github.com/stm32duino/Arduino_Core_STM32.git
git fetch upstream
```
then rebase your master on `remote/upstream/master`
`git rebase -i upstream/master`
Finally remove all unrelated commit and you can also squash the astyle fix.

You can see the [Contributing.md](https://github.com/stm32duino/Arduino_Core_STM32/blob/master/CONTRIBUTING.md) for more details

**Comment by fpistm** [comment]
@jmchiappa 
by squashing your PR, it is ok.
Anyway, a best practice is to create branch base on the master to avoid having all pulled commit in the PR :wink:

**Pull request merged**